plugins/package-managers/node/src/funTest/kotlin/pnpm/PnpmFunTest.kt

```diff
@@ -70,4 +73,14 @@ class PnpmFunTest : StringSpec({
         patchActualResult(result.toYaml(), patchStartAndEndTime = true) should
             matchExpectedResult(expectedResultFile, definitionFile)
     }
+
+    "Resolve dependencies correctly for a nested project" {
+        val definitionFile = getAssetFile("projects/synthetic/pnpm/nested-project/package.json")
+        val result = create("PNPM").collateMultipleProjects(definitionFile)
+
+        result.projects.map { it.id } shouldContainExactlyInAnyOrder listOf(
+            Identifier("PNPM::pnpm-nested-sub-project:1.0.0"),
+            Identifier("PNPM::pnpm-package-with-nested-project:1.0.0")
+        )
```

Comment:
Can we make the test similar to the `Resolve workspace dependencies correctly` in `YarnFunTest`, so that it uses an entire expected result file.

Reply:
(If the approach is to ignore the entries for the nested project, I'm ok to not add this test)

Reply:
The test has been changed.